api/views/workflow.py

```diff
@@ -35,7 +35,7 @@ class SourceViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = SourceSerializer
 
     def get_queryset(self):
-        queryset = Source.objects.all()
+        queryset = Source.objects.exclude(icon='')
```

Comment:
It looks like django will turn this into the correct query? But not with a `.filter`, odd.

```python
In [9]: str(Source.objects.filter(icon='').query)
Out[9]: 'SELECT "share_source"."id", "share_source"."name", "share_source"."long_title", "share_source"."home_page", "share_source"."icon", "share_source"."user_id" FROM "share_source" WHERE "share_source"."icon" = '

In [10]: str(Source.objects.exclude(icon='').query)
Out[10]: 'SELECT "share_source"."id", "share_source"."name", "share_source"."long_title", "share_source"."home_page", "share_source"."icon", "share_source"."user_id" FROM "share_source" WHERE NOT ("share_source"."icon" =  AND "share_source"."icon" IS NOT NULL)'
```

Actually looked at the DB, thought that icon was going to be FK not a string.

@aaxelb's suggest is most correct, I think.